### PR TITLE
feat: Cascade delete option is ticked by default (#3205)

### LINKED ISSUE #3205: cascade delete option is ticked by default , want to have it unticked by default for us.
When you delete app in ARGOCD cascade delete option is by default ticked which can be fatal as it delete resources too from the cluster . Do anyone know how to make default untick when you delete the app from ArgoCD and then choose cascade deletion ok if we really want resources from the cluster to be deleted too.
We use apps-in-apps and want this to be disabled by default.

### PR DESCRIPTION
**Pull request by keithchong**
Signed-off-by: Keith Chong <kykchong@redhat.com>

**Changes from codegen** included.

Fixes https://github.com/argoproj/argo-cd/issues/3205

The test was updated to add the `--force` option simply to test that it is a valid flag. The test does not run with a terminal so it will still pass without it.  This PR requires doc updates to: https://argoproj.github.io/argo-cd/user-guide/commands/argocd_app_delete/ , https://argoproj.github.io/argo-cd/user-guide/app_deletion/
The docs will be updated in a subsequent commit/PR, after this PR has been reviewed, since the doc changes depend on this PR review.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by keithchong** [comment]
Hi @alexmt / @jessesuen , could you please review this?

**Comment by jessesuen** [comment]
I actually feel cascaded deletion is correct in being the default, as this is also the default behavior of kubectl.

Second, won't this break backwards compatibility? Users who are currently relying on `argocd app delete XXXX` will suddenly be left with orphaned resources in the cluster if the default is now to *not* cascade delete.

That said, I do agree that it is quite easy to delete an entire application in the UI, but I believe we've discussed improving the UX to make this less error prone.

**Review by jessesuen — CHANGES_REQUESTED** [review]
Don't think non-cascaded delete should be defaulted. But agree it should be less error prone.

File: docs/user-guide/commands/argocd_app_delete.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ argocd app delete APPNAME [flags]
 
 ```
       --cascade   Perform a cascaded deletion of all application resources (default true)
+      --force     Turn off prompting to confirm cascaded deletion of application resources
```
Comment (jessesuen) [inline docs/user-guide/commands/argocd_app_delete.md:17]: I think we should reserve the word `--force` for something else. (e.g. is there a `kubectl delete --force` that this should match up with instead?

Comment (jsoref) [reply docs/user-guide/commands/argocd_app_delete.md:17]: Yeah, this is odd.

Why can't the `--cascade` flag change to `default=false`?

**Pull request closed without merging**